### PR TITLE
Dependabot upgrades for week of 10th June

### PR DESCRIPTION
**Pull request by DeanElliott96**
## Description of change

Dependabot upgrades for week of 10th June.

Also updated some other babel dependencies as updated are available.

## Checklist

- [x] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

---
Bump @babel/runtime from 7.24.6 to 7.24.7 in /test/sandbox

Bumps [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-runtime)

---
updated-dependencies:
- dependency-name: "@babel/runtime" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/plugin-syntax-import-assertions in /test/sandbox

Bumps [@babel/plugin-syntax-import-assertions](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-syntax-import-assertions) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-plugin-syntax-import-assertions)

---
updated-dependencies:
- dependency-name: "@babel/plugin-syntax-import-assertions" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/core from 7.24.6 to 7.24.7 in /test/sandbox

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-core)

---
updated-dependencies:
- dependency-name: "@babel/core" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/plugin-transform-class-properties from 7.24.6 to 7.24.7

Bumps [@babel/plugin-transform-class-properties](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-transform-class-properties) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-plugin-transform-class-properties)

---
updated-dependencies:
- dependency-name: "@babel/plugin-transform-class-properties" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/plugin-transform-object-rest-spread from 7.24.6 to 7.24.7

Bumps [@babel/plugin-transform-object-rest-spread](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-transform-object-rest-spread) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-plugin-transform-object-rest-spread)

---
updated-dependencies:
- dependency-name: "@babel/plugin-transform-object-rest-spread" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump uuid from 9.0.1 to 10.0.0

Bumps [uuid](https://github.com/uuidjs/uuid) from 9.0.1 to 10.0.0.
- [Changelog](https://github.com/uuidjs/uuid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/uuidjs/uuid/compare/v9.0.1...v10.0.0)

---
updated-dependencies:
- dependency-name: uuid dependency-type: direct:production update-type: version-update:semver-major ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/preset-react from 7.24.6 to 7.24.7

Bumps [@babel/preset-react](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-react) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-preset-react)

---
updated-dependencies:
- dependency-name: "@babel/preset-react" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/eslint-parser from 7.24.6 to 7.24.7

Bumps [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/eslint/babel-eslint-parser)

---
updated-dependencies:
- dependency-name: "@babel/eslint-parser" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump image-minimizer-webpack-plugin from 4.0.1 to 4.0.2

Bumps [image-minimizer-webpack-plugin](https://github.com/webpack-contrib/image-minimizer-webpack-plugin) from 4.0.1 to 4.0.2.
- [Release notes](https://github.com/webpack-contrib/image-minimizer-webpack-plugin/releases)
- [Changelog](https://github.com/webpack-contrib/image-minimizer-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/image-minimizer-webpack-plugin/compare/v4.0.1...v4.0.2)

---
updated-dependencies:
- dependency-name: image-minimizer-webpack-plugin dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump nyc from 15.1.0 to 17.0.0

Bumps [nyc](https://github.com/istanbuljs/nyc) from 15.1.0 to 17.0.0.
- [Release notes](https://github.com/istanbuljs/nyc/releases)
- [Changelog](https://github.com/istanbuljs/nyc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/istanbuljs/nyc/compare/v15.1.0...nyc-v17.0.0)

---
updated-dependencies:
- dependency-name: nyc dependency-type: direct:development update-type: version-update:semver-major ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/eslint-parser from 7.24.6 to 7.24.7 in /test/sandbox

Bumps [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/eslint/babel-eslint-parser)

---
updated-dependencies:
- dependency-name: "@babel/eslint-parser" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

Bump @babel/plugin-transform-optional-chaining from 7.24.6 to 7.24.7

Bumps [@babel/plugin-transform-optional-chaining](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-transform-optional-chaining) from 7.24.6 to 7.24.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.7/packages/babel-plugin-transform-optional-chaining)

---
updated-dependencies:
- dependency-name: "@babel/plugin-transform-optional-chaining" dependency-type: direct:development update-type: version-update:semver-patch ...

Signed-off-by: dependabot[bot] <support@github.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -151,20 +151,20 @@
...
     "@babel/helper-call-delegate": "^7.12.13",
     "@babel/plugin-syntax-import-assertions": "^7.24.6",
-    "@babel/plugin-transform-class-properties": "^7.24.6",
+    "@babel/plugin-transform-class-properties": "^7.24.7",
     "@babel/plugin-transform-nullish-coalescing-operator": "^7.24.6",
-    "@babel/plugin-transform-object-rest-spread": "^7.24.6",
-    "@babel/plugin-transform-optional-chaining": "^7.24.6",
+    "@babel/plugin-transform-object-rest-spread": "^7.24.7",
+    "@babel/plugin-transform-optional-chaining": "^7.24.7",
     "@babel/preset-env": "^7.24.6",
```
Comment (oliverjwroberts) [inline package.json:166]: I think `@babel/preset-env` should be the same version as `@babel/core` where possible (See `docs/Dependabot.md`). Also, I'm not sure where this came from but I thought all `@babel/...` packages need to be the same version? I appreciate that's not the case at the moment on `main`.

**Review by oliverjwroberts — COMMENTED** [review]
Looks good to me. However, before I approve, just two questions...

File: package.json (hunk 1 above)
Comment (DeanElliott96) [reply package.json:166]: 🤔 I manually updated `@babel/core` because I thought`@babel/preset-env` was updated but I must have got mixed up! I'll update!

Comment (DeanElliott96) [reply package.json:166]: Updated now! 😄

Comment (oliverjwroberts) [reply package.json:166]: @DeanElliott96, could both `@babel/core` and `@babel/preset-env` be updated to `7.24.7`?

**Review by oliverjwroberts — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (DeanElliott96) [reply package.json]: Related packages now updated 7.24.7.

**Review by cgsunkel — APPROVED** [review]

**Pull request merged**